pymeasure/instruments/advantest/advantestR624X.py

```diff
@@ -0,0 +1,1040 @@
+        else:
+            lo_relay = int(lo_relay == True) + 1
+
+        self.write(f'ltl 0,{enable},{lo_relay}')
+        self.check_errors()
+
+    def parse_measurement(self, measurement):
+        # TODO, parse header and convert to object      
+        if ' ' in measurement:    # split, and skip header
+            measurement = measurement.split(' ')
+            return (float(measurement[1]), measurement[0])
+        else:
+            return (float(measurement), None)
+
+class AdvantestR6245(AdvantestR624X):
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(
+            adapter,
+            "Advantest R6245 SourceMeter",
+            **kwargs
+        )
+        self.ChA = Channel(self, 1)
+        self.ChB = Channel(self, 2)
+
+class AdvantestR6246(AdvantestR624X):
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(
+            adapter,
+            "Advantest R6246 SourceMeter",
+            **kwargs
+        )
+        self.ChA = Channel(self, 1)
+        self.ChB = Channel(self, 2)
+
+class Channel:
+
+    def __init__(self, instrument, channel):
+        self.instrument = instrument
+        self.channel = channel
+
+    def write(self, command):
+        """ Writes the command to the instrument.
+
+        :param command: command string to be sent to the instrument
+        """
+        self.instrument.write(command)
+
+    def check_errors(self):
+        return self.instrument.check_errors()
+
+    def clear_measurement_buffer(self):
+        """ Clear the measurement data buffer """
+        self.instrument.write(f'mbc {self.channel}')
+        self.check_errors()
+
+    def set_output_type(self, output_type: OutputType, measurement_type: MeasurementType):
+        """ Sets the output method and output data type of output data output by GPIB.
+        """
+
+        output_type = OutputType(output_type)
+        measurement_type = MeasurementType(measurement_type)
+        self.instrument.write(f'ofm {self.channel},{output_type.value},{measurement_type.value}')
+        self.check_errors()
+
+    def set_analog_input(self, value):
+        """ This command turns the analog input terminal (ANALOG INPUT) on the rear panel ON / OFF.
+
+        1. Turn off the analog input.
+        2. Analog input ON, gain x1.
+        3. Analog input ON, gain x2.5.
+        """
+        self.instrument.write(f'ian {self.channel},{value}')
+        self.check_errors()
+
+    def set_trigger_output_timing(self, mask):
+        """ This command sets the timing of the trigger output signal
+        output from TRIGGER OUT on the rear panel.
+
+        b5: 32 At the end of the sweep
+        b4: 16 At the end of the pulse width
+        b3: 8 At the end of the pulse cycle
+        b2: 4 At the end of measurement
+        b1: 2 At the start of measurement
+        b0: 1 At the start of occurrence
+        """
+        self.instrument.write(f'tot {self.channel},{mask}')
+        self.check_errors()
```

Comment:
You can use `Instrument.control` (or equivalent `Channel.control`) for channels. See my above linked documentation regarding channels.

Reply:
Changed accordingly